### ISSUE TITLE
Github Files changed pages are slow

### ISSUE THREAD

**Issue by mdinger** [servo/servo#4810]
Big [github _Files changed_](https://github.com/servo/servo/pull/4719/files) pages lock up every browser including servo. If you can make it usable quicker, it's a big win for servo.

Didn't look for a reduced testcase.

**Comment by mdinger** [comment]
I think servos been trying to open that page for more than 5 minutes.

**Comment by mdinger** [comment]
Firefox and servo both open this page now. Both seem to slow down if the page is scrolled fast but it's usable. Faster would still be nicer.

**Comment by metajack** [comment]
I wonder if the truncated display list patch would help with this. @pcwalton thoughts?

**Comment by lumiscosity** [comment]
The page has since been redesigned to load the changes incrementally and collapse large diffs. Both Firefox and Servo load it pretty fast now, but it doesn't seem like Servo can get past the incremental loader - errors in the console cite not having `IntersectionObserver`, `ResizeObserver` and `AbortController` - not to mention how it struggles with parts of the layout.

**Comment by jdm** [comment]
Using `--enable-experimental-web-platform-features --pref dom_abort_controller_enabled` can help there.

**Comment by lumiscosity** [comment]
Yup, with those flags the errors are gone, and everything's loaded fine - there was a brief lag spike for two seconds, but everything worked fine after it and all the data was loaded. The stylesheet exploded, though, and the console was flooded with `[script::dom::bindings::error] Error at https://github.githubassets.com/assets/vendors-node_modules_oddbird_popover-polyfill_dist_popover-fn_js-a8c266e5f126.js:63:217 e.adoptedStyleSheets is undefined`. Resizing the window takes only a second to re-layout, though!

<img width="1165" height="899" alt="Image" src="https://github.com/user-attachments/assets/884836c7-9c96-4c70-bfe3-295553d3bed0" />

**Comment by jdm** [comment]
Good thing `dom_adoptedstylesheet_enabled` exists as well 😅

**Comment by lumiscosity** [comment]
Geez, really waiting on that prefs page proposal... The page looks the same with it on, though. Only errors left are:

```
[2025-07-26T19:48:14Z ERROR script::dom::bindings::error] Error at :0:0 SyntaxError: The string did not match the expected pattern.
[2025-07-26T19:48:14Z ERROR script::dom::bindings::error] Error at :0:0 SyntaxError: The string did not match the expected pattern.
No valid entry type provided to observe().
[2025-07-26T19:48:15Z ERROR script::dom::bindings::error] Error at :0:0 SyntaxError: The string did not match the expected pattern.
```

**Comment by jdm** [comment]
Alright, let's call this one done, and we can file other issues in the future about specific layout issues. Thanks!
